### ISSUE TITLE
switch freeze randomly

### ISSUE THREAD

**Issue by veda95110** [Kevoot/LaiNESwitch#2]
hi, i have a big issue with laines, the emulator freeze and the switch stop to responding after loading a game, or after switching games...sometime immédiatly sometime after playing 2 or 3 minute, no matter if the switch is docked or not...
I use sx os 1.3 with hb menu 2.0 and my sd is formated in exfat, no auto rcm.
If you have an idea that can be great cause i hope to replay to some old nes game...
Thank you.

**Comment by Kevoot** [comment]
I don't have the same issues, but do not use sx. 

I don't support (nor do I have plans to) debug or support LaiNES for use with the sx platform. If you would like to try it without sx, and the issue still occurs, please let me know in the next few days or I'll close this.

**Comment by veda95110** [comment]
yes that's a good idea i'll give a try with a different system, do you have a suggest?

**Comment by Kevoot** [comment]
Info to run without sx is here: https://gbatemp.net/threads/rcm-payload-hekate-ctcaer-mod.502604/

**Comment by veda95110** [comment]
Thank you i'll try !

Le jeu. 12 juil. 2018 à 21:49, Kevoot <notifications@github.com<mailto:notifications@github.com>> a écrit :

Info to run with sx is here: https://gbatemp.net/threads/rcm-payload-hekate-ctcaer-mod.502604/

**Comment by fincs** [comment]
I also have the same issue. My Switch is on 5.1.0, and I'm using a minimal Hekate setup with just loader and sm.

**Comment by veda95110** [comment]
hello, i found a fix, i changed hbmenu by one find in a headhake distribution (sdfilev52)
Now it's working awesome!!! Thanks for your work Kevoot !
[hbmenu.nro.zip](https://github.com/Kevoot/LaiNESwitch/files/2195144/hbmenu.nro.zip)

**Comment by fincs** [comment]
I compiled hbmenu with latest stable libnx. I don't think there's a reason why it would be the culprit. hbmenu is just a graphical shell that sits on top of hbloader.

**Comment by Kevoot** [comment]
I just tested with the latest hbmenu compiled with libnx and I see now the freezes. Are there points in an particular games that consistently freeze?

**Comment by veda95110** [comment]
for me it's ramdom, sometime it's only when i change games 2 or 3 times, and sometimes it's happen immediatly after i launch the first game on the 2 or 3 first seconds of playing

**Comment by dracnortw** [comment]
happens to me every time when playing zelda 1 just walking around the overworld for a few minutes

**Comment by Kevoot** [comment]
Tracking a memory leak now, valgrind din't pick anything up but hopefully I'll find the source and patch it relatively soon.

**Comment by Kevoot** [comment]
New release uploaded. No crashes when testing with Zelda, played for 20 minutes without issue. Let me know if any (compatible) games crash and if I don't hear anything I'll close this.

**Comment by fincs** [comment]
I tried the latest release. Now I get this on startup:

```
Atmosphère Crash Report (v1.0):
Result:                          0x2A8 (2168-0001)

Process Info:
    Process Name:                hbloader
    Title ID:                    010000000000100d
    Process ID:                  0000000000000084
    Process Flags:               000000b3
    User Exception Address:      00000000556eb000
Exception Info:
    Type:                        Instruction Abort
    Address:                     00000005ef603c5c
Crashed Thread Info:
    Thread ID:                   00000000000005c6
    Stack:                       0000000008000000-0000000008005000
    Registers:
        X[00]:                   0000000000000000
        X[01]:                   0000000000000000
        X[02]:                   0000000000000000
        X[03]:                   0000000000000000
        X[04]:                   0000000000000000
        X[05]:                   0000000000000000
        X[06]:                   0000000000000000
        X[07]:                   0000000000000000
        X[08]:                   0000000000000000
        X[09]:                   0000000000000000
        X[10]:                   0000000000000000
        X[11]:                   0000000000000000
        X[12]:                   0000000000000000
        X[13]:                   0000000000000000
        X[14]:                   0000000000000000
        X[15]:                   0000000000000000
        X[16]:                   0000000000000000
        X[17]:                   0000000000000000
        X[18]:                   0000000000000000
        X[19]:                   00000005ef7b5000
        X[20]:                   0000000bb9635b20
        X[21]:                   0000000bb9635b28
        X[22]:                   00000000000003e8
        X[23]:                   0000000bb9635a40
        X[24]:                   00000005ef5310f4
        X[25]:                   00000005ef7b2000
        X[26]:                   0000000bba179080
        X[27]:                   0000000bb9635960
        X[28]:                   0000000000001000
        FP:                      0000000008004ed0
        LR:                      00000005ef6003d8
        SP:                      0000000008004ec0
        PC:                      00000005ef603c5c
    Stack Trace:
        ReturnAddress[00]:       00000005ef5c5714
        ReturnAddress[01]:       00000005ef59b800
        ReturnAddress[02]:       00000005ef5b201c
        ReturnAddress[03]:       00000005ef5fdddc
        ReturnAddress[04]:       0000000000000000
Code Region Info:
    Number of Code Regions:      0

Thread Report:
Number of Threads:               02
Threads[00]:
    Thread ID:                   00000000000005be
    Stack:                       0000000075ef7000-0000000075ff7000
    Registers:
        X[00]:                   0000000000000000
        X[01]:                   0000000000000000
        X[02]:                   0000000000000000
        X[03]:                   0000000000000000
        X[04]:                   0000000000000000
        X[05]:                   0000000000000000
        X[06]:                   0000000000000000
        X[07]:                   0000000000000000
        X[08]:                   0000000000000000
        X[09]:                   0000000000000000
        X[10]:                   0000000000000000
        X[11]:                   0000000000000000
        X[12]:                   0000000000000000
        X[13]:                   0000000000000000
        X[14]:                   0000000000000000
        X[15]:                   0000000000000000
        X[16]:                   0000000000000000
        X[17]:                   0000000000000000
        X[18]:                   0000000000000000
        X[19]:                   0000000000000000
        X[20]:                   0000000000000000
        X[21]:                   0000000000000000
        X[22]:                   0000000000000000
        X[23]:                   0000000000000000
        X[24]:                   0000000000000000
        X[25]:                   0000000000000000
        X[26]:                   0000000000000000
        X[27]:                   0000000000000000
        X[28]:                   0000000000000000
        FP:                      0000000075ff6e20
        LR:                      0000000060a03fb4
        SP:                      0000000075ff6e20
        PC:                      0000000060a04f6c
    Stack Trace:
        ReturnAddress[00]:       0000000060a01d68
        ReturnAddress[01]:       0000000060a095ec
        ReturnAddress[02]:       0000000060a08544
        ReturnAddress[03]:       0000000060a083d0
        ReturnAddress[04]:       0000000060a06310
        ReturnAddress[05]:       0000000060a00844
        ReturnAddress[06]:       00000005ef604590
        ReturnAddress[07]:       0000000060a08544
        ReturnAddress[08]:       0000000060a083d0
        ReturnAddress[09]:       0000000060a06310
        ReturnAddress[10]:       0000000060a00844
        ReturnAddress[11]:       00000005ef604590
        ReturnAddress[12]:       0000000060a08544
        ReturnAddress[13]:       0000000060a083d0
        ReturnAddress[14]:       0000000060a06310
        ReturnAddress[15]:       0000000060a00844
        ReturnAddress[16]:       00000005ef604590
        ReturnAddress[17]:       0000000060a08544
        ReturnAddress[18]:       0000000060a083d0
        ReturnAddress[19]:       0000000060a06310
        ReturnAddress[20]:       0000000060a00844
        ReturnAddress[21]:       00000005ef604590
        ReturnAddress[22]:       0000000060a08544
        ReturnAddress[23]:       0000000060a083d0
        ReturnAddress[24]:       0000000060a06310
        ReturnAddress[25]:       0000000060a00844
        ReturnAddress[26]:       00000005ef604590
        ReturnAddress[27]:       0000000060a08544
        ReturnAddress[28]:       0000000060a083d0
        ReturnAddress[29]:       0000000060a06310
        ReturnAddress[30]:       0000000060a00844
        ReturnAddress[31]:       00000005ef604590
Threads[01]:
    Thread ID:                   00000000000005c6
    Stack:                       0000000008000000-0000000008005000
    Registers:
        X[00]:                   0000000000000000
        X[01]:                   0000000000000000
        X[02]:                   0000000000000000
        X[03]:                   0000000000000000
        X[04]:                   0000000000000000
        X[05]:                   0000000000000000
        X[06]:                   0000000000000000
        X[07]:                   0000000000000000
        X[08]:                   0000000000000000
        X[09]:                   0000000000000000
        X[10]:                   0000000000000000
        X[11]:                   0000000000000000
        X[12]:                   0000000000000000
        X[13]:                   0000000000000000
        X[14]:                   0000000000000000
        X[15]:                   0000000000000000
        X[16]:                   0000000000000000
        X[17]:                   0000000000000000
        X[18]:                   0000000000000000
        X[19]:                   00000005ef7b5000
        X[20]:                   0000000bb9635b20
        X[21]:                   0000000bb9635b28
        X[22]:                   00000000000003e8
        X[23]:                   0000000bb9635a40
        X[24]:                   00000005ef5310f4
        X[25]:                   00000005ef7b2000
        X[26]:                   0000000bba179080
        X[27]:                   0000000bb9635960
        X[28]:                   0000000000001000
        FP:                      0000000008004ed0
        LR:                      00000005ef6003d8
        SP:                      0000000008004ec0
        PC:                      00000005ef603c5c
    Stack Trace:
        ReturnAddress[00]:       00000005ef5c5714
        ReturnAddress[01]:       00000005ef59b800
        ReturnAddress[02]:       00000005ef5b201c
        ReturnAddress[03]:       00000005ef5fdddc
        ReturnAddress[04]:       0000000000000000
```

**Comment by Kevoot** [comment]
Are you guys following the same directory structure as in the release archive? The .nro and res folder should be located at sdmc:/switch/LaiNes/

This error looks like it couldn't find the res folder

**Comment by Kevoot** [comment]
Oh ffs. Just looked at my code. I forgot to include the "switch" portion of the filepath. Will have to fix when i get home. For now, move the res folder to the root of the sdcard and keep the .nro folder where it is. Should load properly after that.

**Comment by fincs** [comment]
You shouldn't rely on absolute paths. You should just use "res/" and something like "LaiNES.settings" - libnx's default cwd is the folder which contains the nro.

**Comment by Kevoot** [comment]
I'm aware, at one point relative paths were broken and the reason the strings were changed were to accomodate for this. Will be switching back after testing that issue was fixed.

**Comment by fincs** [comment]
Yes, it was fixed in switchbrew/libnx@aa3c35e6e36492d

**Comment by fincs** [comment]
Please now recompile with stable devkitA64 r12 and libnx 1.3.1; and then reupload fixed binaries.
